docs/mcd.md

```diff
@@ -0,0 +1,329 @@
+
+### `<selection-group>`
+
+A required child of field when value-type is `option`. This is the container element for the options that will appear in the drop-down. It has no XML attributes. 
+
+Multiple selection-group elements can be used; see `<conditions>`.
+
+### `<option>`
+
+A child of selection-group, this represents one entry in the drop-down. It has the following XML attributes. 
+
+| Name  | Meaning | Optional? | Value Notes | Other Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| value | The value sent to ConnectionBuilder() | No | | |
+| label | The text the user sees | No | | |
+
+### `<conditions>`
+
+An optional child of field or selection group, this is used for conditional display. It specifies whether the field or selection group should be visible in the dialog based on the values the user has specified for other fields. 
+
+Multiple condition elements can be used, and will be OR'd. That is, the field or selection group will be visible when any one of the `<condition>` elements is matched.
+
+Hence, this is a container element for the condition elements. It has no XML attributes. 
+
+### `<condition>`
+
+| Name  | Meaning | Optional? | Value Notes | Other Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| field | Name of the field whose value will be checked for equality | No | | |
+| value | The value to match. If the field's value is equal to this, the condition is true. | No | | |
+
+
+## Example
+
+The Connection Fields file below demonstrates both conditional display of fields and a non-editable field. 
+
+The images below show the Connection Dialog produced using this file. The left side shows the dialog as it is initially displayed, and the right side shows what it changes to when the user selects Authentication Kerberos. 
+
+<p float="left">
+  <img src="{{ site.baseurl }}/assets/mcd-connection-dialog-1.png" alt="Connection Dialog as initially displayed" align=top hspace=40/>
+  <img src="{{ site.baseurl }}/assets/mcd-connection-dialog-2.png" alt="Connection Dialog after user selects Authentication Kerberos" align=top />
+</p>
+
+AutoReconnect is an example of a non-editable field. It will not be visible in the connection dialog, but its default value of `0` will be available in ConnectionBuilder() to add to the Connection String.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+ 
+<connection-fields>
+
+  <field name="server" label="Server" category="endpoint" value-type="string">
+    <validation-rule reg-exp="^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"/>
+  </field>
+ 
+  <field name="port" label="Port" category="endpoint" value-type="string" default-value="1000"/>
+ 
+  <field name="authentication-type" label="Authentication" category="authentication" value-type="selection" >
+    <selection-group>
+      <option value="0" label="No Authentication"/>
```

Comment:
Can we use any value ? For example instead of value = "0" use value = "Zero" or it is supposed to be from a particular set of valid values ?

Reply:
Yes, any string value. I updated the "Value Notes" tables for option and condition accordingly.